src/app.jsx

```diff
@@ -50,15 +50,22 @@ export const App = ({ name }) => {
     const libvirtService = useObject(() => service.proxy(name), null, []);
     const prevLibvirtServiceRef = useRef();
     const [loadingResources, setLoadingResources] = useState(false);
+    const [error, setError] = useState('');
 
     useEvent(libvirtService, "changed", () => {
         store.dispatch(updateLibvirtState({
             activeState: libvirtService.state,
         }));
         if (libvirtService.state == 'running' && prevLibvirtServiceRef.current !== 'running' && prevLibvirtServiceRef.current !== null) {
             setLoadingResources(true);
+            // If promises rejected here show an toast alert in the main page
             getApiData({ connectionName: 'system' })
-                    .catch(exc => console.warn("getApiData failed:", exc.message))
+                    .then(promises => {
+                        promises.forEach(promise => {
+                            if (promise.status == 'rejected' && promise.reason)
+                                setError(promise.reason.message + '\n' + error);
```

Comment:
I think you want a `findFirst()` or so here, otherwise you call setState() multiple times in quick succession? (My initial reaction was `break`, but of course this is not a proper loop)

Reply:
Fixed.